frontend/public/components/secrets/create-secret/SecretSubForm.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import {
+  SourceSecretForm,
+  SecretTypeAbstraction,
+  PullSecretForm,
+  WebHookSecretForm,
+  GenericSecretForm,
+  SecretType,
+} from '.';
+
+export const SecretSubForm: React.FC<SecretSubFormProps> = ({ typeAbstraction, ...props }) => {
+  switch (typeAbstraction) {
+    case SecretTypeAbstraction.source:
+      return <SourceSecretForm {...props} />;
+    case SecretTypeAbstraction.image:
+      return <PullSecretForm {...props} />;
+    case SecretTypeAbstraction.webhook:
+      return <WebHookSecretForm {...props} />;
+    default:
+      return <GenericSecretForm {...props} />;
+  }
+};
+
+type SecretSubFormProps = {
+  typeAbstraction: SecretTypeAbstraction;
+  onChange: (stringData: SecretStringData) => void;
+  onError: (error: any) => void;
+  onFormDisable: (disable: boolean) => void;
+  stringData: SecretStringData;
+  secretType: SecretType;
+  isCreate: boolean;
+};
+
+type SecretStringData = { [key: string]: string };
```

Comment:
We could share the prop types defined here for the 4 child components if we define `typeAbstraction` in a separate type.
```suggestion
export const SecretSubForm: React.FC<WithTypeAbstraction<SecretSubFormProps>> = ({ typeAbstraction, ...props }) => {
  switch (typeAbstraction) {
    case SecretTypeAbstraction.source:
      return <SourceSecretForm {...props} />;
    case SecretTypeAbstraction.image:
      return <PullSecretForm {...props} />;
    case SecretTypeAbstraction.webhook:
      return <WebHookSecretForm {...props} />;
    default:
      return <GenericSecretForm {...props} />;
  }
};

type WithTypeAbstraction<T> = T & {
  typeAbstraction: SecretTypeAbstraction;
}

export type SecretSubFormProps = {
  typeAbstraction: SecretTypeAbstraction;
  onChange: (stringData: SecretStringData) => void;
  onError: (error: any) => void;
  onFormDisable: (disable: boolean) => void;
  stringData: SecretStringData;
  secretType: SecretType;
  isCreate: boolean;
};

type SecretStringData = { [key: string]: string };
```